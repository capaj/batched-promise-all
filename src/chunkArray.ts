export function chunkArray(array: any[], chunkSize: number) {
  const resultArray: any[] = []
  for (let i = 0, len = array.length; i < len; i += chunkSize) {
    resultArray.push(array.slice(i, i + chunkSize))
  }
  return resultArray
}
