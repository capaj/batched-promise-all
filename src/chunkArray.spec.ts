import { chunkArray } from './chunkArray'

describe('chunk array', () => {
  it('should chunk', async () => {
    expect(chunkArray([1, 2, 3, 4, 5, 6, 7, 8, 9, 0], 3)).toMatchSnapshot()
  })
  it('should not fail on empty array', async () => {
    expect(chunkArray([], 3)).toMatchSnapshot()
  })
})
