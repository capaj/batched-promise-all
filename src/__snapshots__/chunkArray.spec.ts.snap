// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chunk array should chunk 1`] = `
Array [
  Array [
    1,
    2,
    3,
  ],
  Array [
    4,
    5,
    6,
  ],
  Array [
    7,
    8,
    9,
  ],
  Array [
    0,
  ],
]
`;

exports[`chunk array should not fail on empty array 1`] = `Array []`;
